### PR TITLE
Add RBAC labels for Database Services access

### PR DESCRIPTION
**Pull request by marcoandredinis**
This PR adds a new set of Labels that controls access to the DatabaseService resource.

We changed the `RoleSet.checkAccess` to allow for checks in Resources without Labels.
However adding complexity to this function is something that can bit us in the future.
So, this PR reverts that change and adds a new set of labels. Those labels will control whether the RoleSet has access to DatabaseServices.

Given that this is a new condition that we add to the default `editor` role, we had to create a migration/backfill for the roles that don't have them.
This allow us to migrate the roles created previously and add this new allow condition.

This is the output of a (backport) PR's comment: https://github.com/gravitational/teleport/pull/20427#discussion_r1092867699

### CODE REVIEW

File: lib/auth/permissions.go [hunk 1]
```diff
@@ -376,11 +376,12 @@ func (a *authorizer) authorizeRemoteBuiltinRole(r RemoteBuiltinRole) (*Context,
...
-				Namespaces:       []string{types.Wildcard},
-				NodeLabels:       types.Labels{types.Wildcard: []string{types.Wildcard}},
-				AppLabels:        types.Labels{types.Wildcard: []string{types.Wildcard}},
-				DatabaseLabels:   types.Labels{types.Wildcard: []string{types.Wildcard}},
-				KubernetesLabels: types.Labels{types.Wildcard: []string{types.Wildcard}},
+				Namespaces:            []string{types.Wildcard},
+				NodeLabels:            types.Labels{types.Wildcard: []string{types.Wildcard}},
+				AppLabels:             types.Labels{types.Wildcard: []string{types.Wildcard}},
+				DatabaseLabels:        types.Labels{types.Wildcard: []string{types.Wildcard}},
+				DatabaseServiceLabels: types.Labels{types.Wildcard: []string{types.Wildcard}},
```
Comment (r0mant) [inline lib/auth/permissions.go:383]: What does remote proxy need this for?

File: lib/auth/permissions.go [hunk 2]
```diff
@@ -612,8 +614,9 @@ func definitionForBuiltinRole(clusterName string, recConfig types.SessionRecordi
 			role.String(),
 			types.RoleSpecV6{
 				Allow: types.RoleConditions{
-					Namespaces:     []string{types.Wildcard},
-					DatabaseLabels: types.Labels{types.Wildcard: []string{types.Wildcard}},
+					Namespaces:            []string{types.Wildcard},
+					DatabaseLabels:        types.Labels{types.Wildcard: []string{types.Wildcard}},
+					DatabaseServiceLabels: types.Labels{types.Wildcard: []string{types.Wildcard}},
```
Comment (r0mant) [inline lib/auth/permissions.go:619]: What does database service need this for? Does it every query other database services?

File: lib/services/presets.go [hunk 3]
```diff
@@ -50,7 +50,8 @@ func NewPresetEditorRole() types.Role {
 				},
 			},
 			Allow: types.RoleConditions{
-				Namespaces: []string{apidefaults.Namespace},
+				Namespaces:            []string{apidefaults.Namespace},
+				DatabaseServiceLabels: types.Labels{types.Wildcard: []string{types.Wildcard}},
```
Comment (r0mant) [inline lib/services/presets.go:54]: Wouldn't it make more sense to have this as a part of the "access" role where all other labels are?

Comment (marcoandredinis) [reply lib/services/presets.go:54]: Maybe both :thinking: 

So, the Discover User is the one that should have access to the DatabaseServiceLabels
I think that's part of the `editor` responsibilities (they are adding resources).
However, later on they will test the access, which is part of the `access` responsibilities.

I'll add it to the `access` role as well
Let me know if I should drop it from the `editor` role

File: lib/auth/permissions.go (hunk 1 above)
Comment (marcoandredinis) [reply lib/auth/permissions.go]: It shouldn't :+1: 
Removed

File: lib/auth/permissions.go (hunk 2 above)
Comment (marcoandredinis) [reply lib/auth/permissions.go]: It shouldn't :+1: 
Removed

File: lib/services/presets.go (hunk 3 above)
Comment (r0mant) [reply lib/services/presets.go:55]: Yeah, I think it's similar to querying databases or servers or other resources - for the Discover flow the "access" role is what allows the user to do that, right? So I think let's remove it from here. You still need "access" role to be able to go through the Discover flow so you should have these labels.

**Review by r0mant — APPROVED** [review]
Let's remove db_service_labels from editor role, otherwise lgtm.

File: lib/services/presets.go (hunk 3 above)
Comment (marcoandredinis) [reply lib/services/presets.go:55]: Yup, good point
Removing

**Comment by marcoandredinis** [comment]
@ryanclark @Tener Can you please take a look at this PR? :pray:

**Review by ryanclark — APPROVED** [review]

**Pull request merged**